src/backend/distributed/shared_library_init.c

```diff
@@ -924,6 +925,63 @@ RegisterCitusConfigVariables(void)
 		0,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.check_available_space_before_move",
+		gettext_noop("When enabled will check free disk space before a shard move"),
+		gettext_noop(
+			"Free disk space will be checked when this setting is enabled before each shard move."
+			"See citus.delete_old_shards_sleep_time and citus.delete_old_shards_max_tries"),
+		&CheckAvailableSpaceBeforeMove,
+		true,
+		PGC_USERSET,
```

Comment:
I think we can mark this as NO_SHOW Guc.

Reply:
Don't we want users to see this setting? I think if we simplify the gucs to https://github.com/citusdata/citus/pull/4929#discussion_r623971584 then it might make sense to make this user visible, what do you think? 

Maybe we can make `DesiredPercentFreeAfterMove` invisible to users and decide the value ourselves as Marco suggested. 

So the only user visible setting could be `citus.check_available_space_before_move`, from user point of the view, the interaction might be easier/simpler this way. We can keep `citus.desired_percent_disk_available_after_move` as a GUC so that if we are not okay with the default value we can change it in production.